From 63ee4db73a7047b55c7f7b5badb759a04e9ae00c Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Thu, 9 Oct 2014 21:00:01 -0400
Subject: [fs] xfs: wrong error sign conversion during failed DIO writes

Message-id: <1412888441-4181-8-git-send-email-sandeen@redhat.com>
Patchwork-id: 97313
O-Subject: [RHEL7.1 PATCH 07/47] [fs] xfs: wrong error sign conversion during failed DIO writes
Bugzilla: 1145837
RH-Acked-by: Brian Foster <bfoster@redhat.com>
RH-Acked-by: Dave Chinner <dchinner@redhat.com>

Bugzilla: 1145837
Upstream Status: Committed upstream
Build Info: https://brewweb.devel.redhat.com/taskinfo?taskID=8082426
Tested: locally using xfstests

Thanks,
-Eric

    commit 07d5035a289f8bebe0ea86c293b2d5412478c481
    Author: Dave Chinner <dchinner@redhat.com>
    Date:   Thu Apr 17 08:15:27 2014 +1000

    xfs: wrong error sign conversion during failed DIO writes

    We negate the error value being returned from a generic function
    incorrectly. The code path that it is running in returned negative
    errors, so there is no need to negate it to get the correct error
    signs here.

    This was uncovered by generic/019.

    Signed-off-by: Dave Chinner <dchinner@redhat.com>
    Reviewed-by: Christoph Hellwig <hch@lst.de>
    Signed-off-by: Dave Chinner <david@fromorbit.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/xfs/xfs_file.c b/fs/xfs/xfs_file.c
index 1777ada..f64d01e 100644
--- a/fs/xfs/xfs_file.c
+++ b/fs/xfs/xfs_file.c
@@ -679,7 +679,7 @@ xfs_file_dio_aio_write(
   goto out;
 
  if (mapping->nrpages) {
-  ret = -filemap_write_and_wait_range(VFS_I(ip)->i_mapping,
+  ret = filemap_write_and_wait_range(VFS_I(ip)->i_mapping,
           pos, -1);
   if (ret)
    goto out;
-- 
1.7.1